ToolBox/SECDataDownloader/SECDataConverter.cs

```diff
@@ -123,189 +116,189 @@ public SECDataConverter(string rawSource, string destination, string knownEquity
                         }
                     }
                 );
-
-            KnownEquities = Directory.GetFiles(knownEquityFolder)
-                .Select(x => Path.GetFileNameWithoutExtension(x).ToLower())
-                .ToList();
         }
 
         /// <summary>
         /// Converts the data from raw format (*.nz.tar.gz) to json files consumable by LEAN
         /// </summary>
         /// <param name="startDate">Starting date to start process files</param>
         /// <param name="endDate">Ending date to stop processing files</param>
-        public void Process(DateTime startDate, DateTime endDate)
+        public void Process(DateTime processingDate)
         {
+            var formattedDate = processingDate.ToString(DateFormat.EightCharacter);
+            var remoteRawData = new FileInfo(Path.Combine(RawSource, $"{formattedDate}.nc.tar.gz"));
+            if (!remoteRawData.Exists)
+            {
+                throw new Exception($"SECDataConverter.Process(): Raw data {remoteRawData} not found. No process can be done.");
+            }
+
+            Log.Trace($"SECDataConverter.Process(): Copying raw data locally...");
+
+            var localRawData = remoteRawData.CopyTo(Path.Combine(Path.GetTempPath(), remoteRawData.Name));
+            var extractDataPath = Path.Combine(Path.GetTempPath(), formattedDate);
+
+            Log.Trace($"SECDataConverter.Process(): Extract raw data...");
+
+            using (var data = localRawData.OpenRead())
+            {
+                using (var archive = TarArchive.CreateInputTarArchive(new GZipInputStream(data)))
```

Comment:
I would have expected the `Compression` library to support `.tar.gz`/`.tgz` files. We may want to add this functionality to the `Compression.cs` class so we don't end up duplicating this code elsewhere.